lib/unifex/specs.ex

```diff
@@ -70,26 +70,58 @@ defmodule Unifex.Specs do
     {_res, binds} = Code.eval_string(specs_code, [{:unifex_config__, []}], make_env(specs_file))
     config = binds |> Keyword.fetch!(:unifex_config__) |> Enum.reverse()
 
-    {functions_args, functions_results} =
+    {functions_args_arity, functions_results} =
       config
       |> Keyword.get_values(:function)
-      |> Enum.map(fn {name, args, results} -> {{name, args}, {name, results}} end)
+      |> Enum.map(fn {name, args, results} ->
+        {{{name, args}, {name, Enum.count(args)}}, {name, results}}
+      end)
       |> Enum.unzip()
 
+    {functions_args, functions_arity} = Enum.unzip(functions_args_arity)
+
     functions_results =
       Enum.flat_map(functions_results, fn {name, results} -> Enum.map(results, &{name, &1}) end)
 
     functions_docs = parse_docs(config, specs_file)
 
+    dirty_functions =
+      config
+      |> Keyword.get_values(:dirty_functions)
+      |> List.flatten()
+      |> Enum.map(fn {dirty_func, type} ->
+        {dirty_name, dirty_arity} =
+          cond do
+            is_tuple(dirty_func) ->
+              dirty_func
+
+            is_atom(dirty_func) ->
+              {dirty_func, nil}
+          end
+
+        {matched_name, matched_arity} = List.keyfind(functions_arity, dirty_name, 0, {nil, nil})
+
+        if matched_name == dirty_name and (matched_arity == dirty_arity or dirty_arity == nil) do
```

Comment:
```suggestion
        dirty_name =
          case dirty_func do
            {name, _arity} -> name
            name -> name
          end

        {matched_name, matched_args} = List.keyfind(functions_args, dirty_name, 0, {nil, nil})

        if dirty_func in [matched_name, {matched_name, length(matched_args)}] do
```

slightly more idiomatic, and you don't need `functions_arity`